api/v1alpha1/edgeplaybook_types.go

```diff
@@ -0,0 +1,116 @@
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgePlaybookSpec defines the desired state of EdgePlaybook
+type EdgePlaybookSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	DeviceSelector *metav1.LabelSelector `json:"deviceSelector,omitempty"`
+
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+
+	LogCollection map[string]*LogCollectionConfig `json:"logCollection,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// Ansible binary file
+	BinaryPath string `json:"binaryPath,omitempty"`
+	// the ansible's playbooks list to be used
+	Playbooks []Playbook `json:"playbooks,omitempty"`
+	// the ansible's playbook options
+	Options *AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options
+	PrivilegeEscalationOptions *PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+	// enum that defines which is the stdout callback method (default = 'default'). Supported stdout methods are: debug,default,dense,json,minimal,null,oneline,stderr,timer,yaml,runner-events
+	StdoutCallback string `json:"stdoutCallback,omitempty"`
+}
+
+type AnsibleOptions struct {
+	// don't make any changes; instead, try to predict some of the changes that may occur
+	Check bool `json:"check,omitempty"`
+}
+
+type PrivilegeEscalationOptions struct {
+	Become bool `json:"become,omitempty"`
+	// Enum ksu, pbrun, enable, sesu, pmrun,runas, sudo, su, doas, pfexec, Machinectl, dzdo
+	BecomeMethod string `json:"becomeMethod,omitempty"`
+	BecomeUser   string `json:"becomeUser,omitempty"`
+}
+
+type Playbook struct {
+	// Playbook content
+	Content string `json:"content"`
+	// The connection timeout on ansible-playbook
+	Timeout uint64 `json:"timeout,omitempty"`
+	// The required privelege level necessary to execute the playbook TODO: Enum like linux capabilities ?
+	RequiredPrivilegeLevel string `json:"requiredPrivilegeLevel,omitempty"`
+}
+
+// EdgePlaybookStatus defines the observed state of EdgePlaybook
+type EdgePlaybookStatus struct {
+	Phase                     string               `json:"phase,omitempty"`
+	LastSeenTime              metav1.Time          `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string               `json:"lastSyncedResourceVersion,omitempty"`
+	Hardware                  *Hardware            `json:"hardware,omitempty"`
+	Deployments               []PlaybookDeployment `json:"playbookDeployments,omitempty"`
+	DataOBC                   *string              `json:"dataObc,omitempty"`
+	UpgradeInformation        *UpgradeInformation  `json:"upgradeInformation,omitempty"`
+}
+
+type PlaybookDeployment struct {
+	Name               string              `json:"name"`
```

Comment:
Should the user somehow define the name, so he knows which playbook CR to check for the status?

Reply:
I don't completely get the question. Can you clarify a bit? 

From an `EdgePlaybook` CR, the operator will generate several `PlaybookExecution` CR (one for each device on which the playbook must be run).

Where are you suggesting to put the field `Name`?